src/parseTree.js

```diff
@@ -281,37 +270,121 @@
         return obj;
       },
+      children: function() {
+        this.$data.childrenNodes = this.classObj.collapsed ? undefined : [];
+        var self = this;
+        function enter(node, parent, depth) {
+          // Don't show or recurse into nodes that failed, unless "Show failures" is enabled.
+          if ((!node.succeeded && !ohmEditor.options.showFailures) ||
+              (node.isImplicitSpaces && !ohmEditor.options.showSpaces)) {
+            return node.SKIP;
+          }
+          // Don't bother showing whitespace nodes that didn't consume anything.
+          var isWhitespace = node.expr.ruleName === 'spaces';
+          if (isWhitespace && node.source.contents.length === 0) {
+            return node.SKIP;
+          }
+
+          var traceElement = {
+            traceNode: node,
+            context: {
+              parent: self,
+              collapsed: self.collapsed,
+              syntactic: self.labeled ? isSyntactic(node.expr) :
+                                        self.context && self.context.syntactic,
+              vbox: self.vbox
+            },
+            isInVBox: self.context ? self.context.vbox : false,
+            currentLR: self.currentLR
+          };
+          self.$data.childrenNodes.push(traceElement);
+          return node.SKIP;
+        }
+
+        if (this.$data.childrenNodes) {
+          if (this.traceNode.isRootNode) {
+            this.traceNode.walk(enter);
+          } else {
+            this.traceNode.children.forEach(function(child) {
+              child.walk(enter);
+            });
+          }
+        }
+        return this.$data.childrenNodes;
+      },
       minWidth: function() {
         return this.measureInputText(this.traceNode.source.contents) + 'px';
       }
     },
+    data: function() {
+      return {
+        childrenNodes: undefined
+      };
+    },
     template: [
-      '<div class="pexpr" :class="classObj">',
-      '  <div v-if="classes.labeled" class="self">',
+      '<div class="pexpr" :class="classObj" id="getFreshNodeId()">',
+      '  <div v-if="labeled" class="self">',
       '    <trace-label :traceNode="traceNode" :minWidth="minWidth"',
       '                 @hover="onHover" @unhover="onUnhover" @click="onClick"',
       '                 @showContextMenu="onShowContextMenu" />',
       '  </div>',
-      '  <div v-if="!classes.leaf" ref="children"',
+      '  <div v-if="!isLeaf" ref="children"',
       '       class="children" :class="{vbox: vbox}"',
-      '       :hidden="classes.collapsed">',
-      '    <slot />',
+      '       :hidden="classObj.collapsed">',
+      '    <trace-element :id="child.id" :traceNode="child.traceNode" :context="child.context"',
+      '                   :currentLR="child.currentLR" :measureInputText="measureInputText"',
+      '                   :isInVBox="child.isInVBox" :eventHandlers="eventHandlers"',
+      '                   v-for="child in children"></trace-element>',
```

Comment:
I'd still the `v-for` on the first line. It's a bit weird to read all the references to `child` before seeing the `v-for`.